syntax = "proto3";
import "dtypes.proto";

package formula_parser;

service FormulaParser {
    rpc ParseFormula(FormulaParserRequest) returns (FormulaParserResponse) {}
}

message FormulaParserRequest {
    string formula = 1;  // La fórmula de Excel a parsear
}

message FormulaParserResponse {
    string formula = 1;                  // La fórmula original
    optional dtypes.Tokens tokens = 2;   // Tokens generados durante el parseo
    optional dtypes.AST ast = 3;         // El Árbol de Sintaxis Abstracta resultante
    string error = 4;                    // Mensaje de error si hubo un problema al parsear
}
